src/modules/auth/auth.service.ts

```diff
@@ -86,6 +136,52 @@ export class AuthService {
     return { accessToken, refreshToken };
   }
 
+  async verifyAndCreateUser(token: string) {
+    const pendingUserKeys =
+      await this.redisService.redis.keys('pending_user:*');
+
+    let pendingUserData: any = null;
+    let pendingKey: string | null = null;
+
+    for (const key of pendingUserKeys) {
```

Comment:
why are you looping here and getting all user keys? why not just set data using token and check if it exists by using it as a key when getting.

so just 
```ts
this.redis.service.get(token) // null or string
```